### PR TITLE
Redis

### PR DESCRIPTION
**Pull request by abushev**
- Updated framework version
- Added redis client submodule
- Added redis client adapter and its usage in http_bidder
- With 100% matched request bidder handles 6500 qps on my machine vs 10500 qps when unmatched requests.

### CODE REVIEW

**Comment by abushev** [comment]
Btw how to change framework add-on to clone it without publickey?

**Comment by venediktov** [comment]
We can change .gitmodule to use http instead of git which uses ssh

[submodule "framework"]
        path = framework
        url = "https://github.com/venediktov/vanilla-rtb.git"

File: .gitmodules [hunk 1]
```diff
@@ -1,3 +1,6 @@
 [submodule "framework"]
 	path = framework
 	url = git@github.com:venediktov/vanilla-rtb.git
```
Comment (venediktov) [inline .gitmodules]: I propose to change it to https://  - travis is failing 
url = "https://github.com/venediktov/vanilla-rtb.git"

**Comment by venediktov** [comment]
btw , perhaps the problem with the build is related to directory name /deps ? Because in my travis script I also use directory /deps for dependencies , perhaps they collide ?

**Comment by venediktov** [comment]
CMake Error: The source directory "/home/travis/build/vanilla-rtb/rapid-bidder/build-vanilla/submodules/redisclient" **does not appear to contain CMakeLists.txt.**

**Comment by venediktov** [comment]
let's remove cache in travis and rebuild , perhaps the cache is a problem ?
https://travis-ci.org/vanilla-rtb/rapid-bidder/caches

**Comment by abushev** [comment]
Yeah, because travis doesn't actually updates submodules. Steal no idea why.

File: .gitmodules [hunk 2]
```diff
@@ -1,3 +1,6 @@
 [submodule "framework"]
 	path = framework
-	url = git@github.com:venediktov/vanilla-rtb.git
+	url = https://github.com/venediktov/vanilla-rtb.git
+[submodule "submodules/redisclient"]
+	path = submodules/redisclient
+	url = https://github.com/nekipelov/redisclient
```
Comment (venediktov) [inline .gitmodules:6]: url = https://github.com/nekipelov/redisclient.git

You are missing extension .git  , that's probably why it does not load it in travis ?

CMake Error: The source directory "/home/travis/build/vanilla-rtb/rapid-bidder/build-vanilla/submodules/redisclient" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.
make[2]: *** [submodules/redisclient/src/redisclient-stamp/redisclient-configure] Error 1
make[2]: Leaving directory `/home/travis/build/vanilla-rtb/rapid-bidder/build-vanilla'
make[1]: *** [CMakeFiles/redisclient.dir/all] Error 2
make[1]: *** Waiting for unfinished jobs....

Comment (venediktov) [reply .gitmodules:6]: so it is not a problem with /deps directory , it's just misspelt , if you fix it and rerun it it should build even with /deps directory

File: exchange_server/multiexchange_config.hpp [hunk 3]
```diff
@@ -9,20 +9,27 @@
 #define MULTIEXCHANGE_CONFIG_HPP
 
 #include <string>
-#include "rtb/config/config.hpp"
+#include "../../rtb/config/config.hpp"
```
Comment (venediktov) [inline exchange_server/multiexchange_config.hpp]: it should compile without ../../   -> #include "rtb/config/config.hpp"

**Review by venediktov — COMMENTED** [review]
Травис не находит boost когда собирает radis . Если ты на своём компе временно переменяешь папку с boost , и сделаешь как мы в трависе export CMAKE_OPTIOMS += BOOST_ROOT=.....  направив на новую переименованную папку , то ванила соберётся , а радис воядли . В логе написано что boost он неможет найти . Может и dependency придётся потом добавить , которые нужны radis client

**Comment by venediktov** [comment]
error: wrong library name 'unit_test_framework' in the --with-<library> option.

**Comment by venediktov** [comment]
Смотри всегда raw log , там больше информации об ошибках .

**Comment by abushev** [comment]
Ага, я видел. Вопрос в том почему ее там нет. В стандартной 1.60 она есть.

**Pull request merged**